modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java

```diff
@@ -137,6 +137,58 @@ public void before() {
         .containingClass(echoTestClassFile, "org/foo/EchoTest.class");
   }
 
+  @Test
+  @Ignore("W-16508382: Java 9+ restricts extending standard Java packages like javax.annotation, causing ResolutionException. "
+      + "Replaced by canExtendCustomPackages.")
+  public void canExtendJavaPackages() throws Exception {
+    File customJavaxClassFile = new SingleClassCompiler()
+        .compile(getResourceFile("/javax/annotation/CustomAnnotation.java"));
+
+    // Create an application that includes a custom class in the javax.annotation package
+    final ApplicationFileBuilder extendJavaPackagesAppFileBuilder = appFileBuilder("appExtendingJavaPackages")
+        .definedBy("app-extending-java-packages-config.xml")
+        .containingClass(customJavaxClassFile, "javax/annotation/CustomAnnotation.class");
+
+    addPackedAppFromBuilder(extendJavaPackagesAppFileBuilder);
+    startDeployment();
+    assertDeploymentSuccess(applicationDeploymentListener, extendJavaPackagesAppFileBuilder.getId());
+    executeApplicationFlow("main");
+  }
+
+  @Test
+  @Issue("W-16508382: Cover missing test cases from ClassloadingTestCase by adding tests for external library dependencies and custom package extensions.")
+  public void canExtendCustomPackages() throws Exception {
+    File customExtensionJarFile = new SingleClassCompiler()
+        .compile(getResourceFile("/com/example/extension/CustomExtension.java"));
+
+    // Create an application that includes a custom class in the com.example.extension package
+    final ApplicationFileBuilder extendPackagesAppFileBuilder = appFileBuilder("appExtendingCustomPackages")
+        .definedBy("app-extending-custom-packages-config.xml")
+        .containingClass(customExtensionJarFile, "com/example/extension/CustomExtension.class");
+
+    addPackedAppFromBuilder(extendPackagesAppFileBuilder);
+    startDeployment();
+    assertDeploymentSuccess(applicationDeploymentListener, extendPackagesAppFileBuilder.getId());
+    executeApplicationFlow("main");
+  }
+
+  @Test
+  @Issue("W-16508382: Cover missing test cases from ClassloadingTestCase by adding tests for external library dependencies and custom package extensions.")
+  public void canDependOnExternalLibrary() throws Exception {
+    File externalJarFile = new JarCompiler().compiling(getResourceFile("/com/example/external/ExternalClass.java"))
```

Comment:
The variable name doesn't reflect the content, it's not a JAR but a class

Reply:
removed this test, and added canDependOnXerces test